### PR TITLE
Add SPM `MapboxCoreNavigation` integration test

### PR DESCRIPTION
**Pull request by azarovalex**
Adds a CI build step to make sure that an application builds with only `MapboxCoreNavigation` dependency.

### CODE REVIEW

File: Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import CoreLocation
+
+struct ContentView: View {
+    var body: some View {
+        Button("Request route") {
+            let routeOptions = RouteOptions(coordinates: [
+                CLLocationCoordinate2D(latitude: 10, longitude: 10),
+                CLLocationCoordinate2D(latitude: 11, longitude: 11)
+            ])
+            Directions.shared.calculate(routeOptions) { _, result in
```
Comment (bamx23) [inline Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift:13]: Could you use `calculateWithCache` to check `MapboxCoreNavigation` is imported?

Comment (Udumft) [reply Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift:13]: I would not advise to use exactly `calculateWithCache` as this method most likely will be [removed](https://github.com/mapbox/mapbox-navigation-ios/pull/3261/files#diff-c3b17629de0ab1de692fbf26006d4edeb545b1f47dbd97b63ec55ba686d2b6a4L19) when we implement `HybridRouter`.

Comment (azarovalex) [reply Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift:13]: `RouteOptions` is already a part of `MapboxCoreNavigation`

Comment (azarovalex) [reply Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift:13]: Sorry, Xcode is lying, `RouteOptions` is a part of `Directions`
image

Comment (azarovalex) [reply Tests/SPMTest/CoreSPMTest/CoreSPMTest/ContentView.swift]: I added `NavigationEventsManager` usage, so now we can be sure that MapboxCoreNavigation is linked.

**Review by bamx23 — APPROVED** [review]

**Pull request merged**